wallets/service/wallet_service.py

```diff
@@ -0,0 +1,228 @@
+        self.EXECUTOR_WALLET_ADDRESS = get_ssm_parameter(EXECUTOR_ADDRESS)
+        self.EXECUTOR_WALLET_KEY = get_ssm_parameter(EXECUTOR_KEY)
+        method_name = "openChannelByThirdParty"
+        self.mpe_address = self.obj_blockchain_util.read_contract_address(
+            net_id=NETWORK_ID, path=MPE_ADDR_PATH,
+            key='address'
+        )
+
+        current_block_no = self.obj_blockchain_util.get_current_block_no()
+
+        # 1 block no is mined in 15 sec on average, setting expiration as 10 years
+        expiration = current_block_no + (10 * 365 * 24 * 60 * 4)
+        agi_tokens = self.__calculate_agi_tokens(amount=amount, currency=currency)
+
+        group_id_in_hex = "0x" + base64.b64decode(group_id).hex()
+        r, s, v, signature = self.__generate_signature_details(
+            recipient=recipient, group_id=group_id_in_hex,
+            agi_tokens=agi_tokens, expiration=expiration,
+            message_nonce=current_block_no,
+            signer_key=sender_private_key
+        )
+
+        positional_inputs = (
+            sender, SIGNER_ADDRESS, recipient,
+            group_id_in_hex, agi_tokens, expiration,
+            current_block_no, v, r, s
+        )
+
+        transaction_object = self.obj_blockchain_util.create_transaction_object(
+            *positional_inputs,
+            method_name=method_name,
+            address=self.EXECUTOR_WALLET_ADDRESS,
+            contract_path=MPE_CNTRCT_PATH,
+            contract_address_path=MPE_ADDR_PATH,
+            net_id=NETWORK_ID
+        )
+
+        raw_transaction = self.obj_blockchain_util.sign_transaction_with_private_key(
+            transaction_object=transaction_object,
+            private_key=self.EXECUTOR_WALLET_KEY)
+        transaction_hash = self.obj_blockchain_util.process_raw_transaction(raw_transaction=raw_transaction)
+
+        print("openChannelByThirdParty::transaction_hash", transaction_hash)
+
+        self.channel_dao.insert_channel_history(
+            order_id=order_id, amount=amount, currency=currency,
+            group_id=group_id, org_id=org_id,
+            type=method_name, recipient=recipient,
+            address=sender, signature=signature,
+            request_parameters=str(positional_inputs),
+            transaction_hash=transaction_hash, status=TransactionStatus.PENDING
+        )
+
+        return {
+            "transaction_hash": transaction_hash, "signature": signature,
+            "agi_tokens": agi_tokens, "type": method_name
+        }
+
+    def set_default_wallet(self, username, address):
+        self.obj_wallet_dao.set_default_wallet(username=username, address=address)
+        return "OK"
+
+    def add_funds_to_channel(self, org_id, group_id, channel_id, sender, recipient, order_id, amount, currency):
+        self.EXECUTOR_WALLET_ADDRESS = get_ssm_parameter(EXECUTOR_ADDRESS)
+        self.EXECUTOR_WALLET_KEY = get_ssm_parameter(EXECUTOR_KEY)
```

Comment:
move this as global variable as this will call ssm everytime and that will cost us .

Reply:
I should only call the ssm while performing channel transactions